pkg/database/fake/database_generated_test.go

```diff
@@ -144,3 +144,92 @@ func TestQueryNamespacedResources(t *testing.T) {
 		assert.Equal(t, tt.expected, filteredResources)
 	}
 }
+
+func TestQueryCoreResources(t *testing.T) {
+
+	existingKind := testResources[1].GetKind()
+	existingVersion := testResources[1].GetAPIVersion()
+
+	tests := []struct {
+		name     string
+		kind     string
+		version  string
+		expected []*unstructured.Unstructured
+	}{
+		{
+			name:     "No matching resources by kind",
+			kind:     "NotFound",
+			version:  existingVersion,
+			expected: nil,
+		},
+		{
+			name:     "No matching resources by version",
+			kind:     existingKind,
+			version:  "v2",
+			expected: nil,
+		},
+		{
+			name:     "Matching resources",
+			kind:     existingKind,
+			version:  existingVersion,
+			expected: testResources[1:],
```

Comment:
This makes it error prone if we add a new element to the list. What about recreating the list? The other option is

```suggestion
			expected: []*unstructured.Unstructured{testResources[1]},
```


```suggestion
			expected: testResources[1:1],
```

Reply:
Addressed, but note that I've used `testResources[1:2]` instead of `testResources[1:1]` The last one doesn't do what we expect